File: sms-service/src/test/java/com/iiplabs/nns/sms/utils/StringUtilTest.java
package com.iiplabs.nns.sms.utils;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

public class StringUtilTest {

	@Test
	public void testGoodLastField() {
		String source = "card.expiry";
		assertEquals("expiry", StringUtil.getLastField(source));
	}

	@Test
	public void testBadLastField() {
		String source = "expiry";
		assertEquals("expiry", StringUtil.getLastField(source));
	}

}

File: ksms-service/src/test/kotlin/com/iiplabs/nns/ksms/utils/StringUtilTest.kt
package com.iiplabs.nns.ksms.utils

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

class StringUtilTest {

    @Test
    fun testGoodLastField() {
        val source = "card.expiry"
        assertEquals("expiry", StringUtil.getLastField(source))
    }

    @Test
    fun testBadLastField() {
        val source = "expiry"
        assertEquals("expiry", StringUtil.getLastField(source))
    }

    @Test
    fun testMatchAllBetweenBrackets() {
        val fieldsRegex = "^\\((.*?)\\)$".toRegex()
        assertFalse(fieldsRegex matches "")
        assertFalse(fieldsRegex matches "test(test123)1")
        assertTrue(fieldsRegex matches "()")
        assertTrue(fieldsRegex matches "(123)")
        assertTrue(fieldsRegex matches "(test123test)")
    }
}